GCC 7 dl linker errors

When linking the unit tests with libtiledb, the following linker errors can occur on some systems:

```
[ 59%] Linking CXX executable tiledb_unit
../../externals/src/ep_tbb-build/_release/libtbb.a(dynamic_link.o): In function `tbb::internal::dynamic_unlink(void*)':
/tmp/TileDB/build/externals/src/ep_tbb/./src/tbb/dynamic_link.cpp:420: undefined reference to `dlclose'
../../externals/src/ep_tbb-build/_release/libtbb.a(dynamic_link.o): In function `tbb::internal::dynamic_link(char const*, tbb::internal::dynamic_link_descriptor const*, unsigned long, void**, int)':
/tmp/TileDB/build/externals/src/ep_tbb/./src/tbb/dynamic_link.cpp:478: undefined reference to `dlopen'
/tmp/TileDB/build/externals/src/ep_tbb/./src/tbb/dynamic_link.cpp:480: undefined reference to `dlopen'
/tmp/TileDB/build/externals/src/ep_tbb/./src/tbb/dynamic_link.cpp:126: undefined reference to `dlsym'
```

The workaround is to add the following to `test/CMakeLists.txt`:

```cmake
set_target_properties(tiledb_unit PROPERTIES
  LINK_FLAGS "-Wl,--no-as-needed -ldl"
)
```

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -148,6 +148,13 @@ target_compile_definitions(tiledb_unit PRIVATE
   -DTILEDB_TEST_INPUTS_DIR="${CMAKE_CURRENT_SOURCE_DIR}/inputs"
 )
 
+# Linking dl is only needed on linux with gcc
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    set_target_properties(tiledb_unit PROPERTIES
+      LINK_FLAGS "-Wl,--no-as-needed -ldl"
+    )
+endif()
+
 add_test(
   NAME "tiledb_unit"
   COMMAND $<TARGET_FILE:tiledb_unit> --durations=yes

